src/components/Map/index.js

```diff
@@ -1,18 +1,105 @@
-import React from "react";
-import styled from "styled-components";
+import React, { useEffect, useRef, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+import mapsData from "../../data/mapData";
 
 function Map() {
-  return <MapImage>Map (400px X 400px)</MapImage>;
-}
+  const ref = useRef(null);
+  const [isMapData, setIsMapData] = useState(false);
+  const [map, setMap] = useState([]);
+  const navigate = useNavigate();
+
+  const catSprite = new Image();
+  const assets = new Image();
+
+  catSprite.src = "/assets/image/cat.png";
+  assets.src = "/assets/image/coblocks-assets.png";
+
+  useEffect(() => {
+    if (mapsData) {
+      const key = window.location.pathname.split("/")[2];
```

Comment:
주소에서 gameId를 가져오기 위해, split으로 주소를 나눠주신게 맞을까요?
그렇다면 해당 주소의 `gameId`는 useParams를 사용해서 동적으로 가져올 수 있기 때문에, 아래와 같이 불러오는 방식이 더 명시적인 방법으로 생각됩니다.
```suggestion
const { gameId } = useParams();
```